File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task139.py

import math
import sys

def calculate_circle_area(radius):
    """\n    Calculate the area of a circle given its radius.\n    \n    Args:\n        radius: The radius of the circle (float or int)\n    \n    Returns:\n        The area of the circle\n    \n    Raises:\n        ValueError: If radius is negative or invalid\n    """
    # Input validation
    if not isinstance(radius, (int, float)):
        raise TypeError("Radius must be a numeric value")
    
    if radius < 0:
        raise ValueError("Radius cannot be negative")
    
    if math.isnan(radius) or math.isinf(radius):
        raise ValueError("Invalid radius value")
    
    # Check for potential overflow
    if radius > math.sqrt(sys.float_info.max / math.pi):
        raise OverflowError("Radius too large, calculation would overflow")
    
    # Calculate area
    area = math.pi * radius * radius
    
    return area

def main():
    """Main function with test cases"""
    print("Test Case 1 - Radius: 5.0")
    print(f"Area: {calculate_circle_area(5.0)}")
    
    print("\\nTest Case 2 - Radius: 10.5")
    print(f"Area: {calculate_circle_area(10.5)}")
    
    print("\\nTest Case 3 - Radius: 0.0")
    print(f"Area: {calculate_circle_area(0.0)}")
    
    print("\\nTest Case 4 - Radius: 1.0")
    print(f"Area: {calculate_circle_area(1.0)}")
    
    print("\\nTest Case 5 - Radius: 7.5")
    print(f"Area: {calculate_circle_area(7.5)}")
    
    # Test error case
    try:
        print("\\nTest Case 6 - Radius: -5.0 (Error case)")
        print(f"Area: {calculate_circle_area(-5.0)}")
    except ValueError as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()

File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task139.java

import java.util.Scanner;

public class Task139 {
    private static final double PI = 3.14159265358979323846;
    
    public static double calculateCircleArea(double radius) {
        // Input validation
        if (radius < 0) {
            throw new IllegalArgumentException("Radius cannot be negative");
        }
        if (Double.isNaN(radius) || Double.isInfinite(radius)) {
            throw new IllegalArgumentException("Invalid radius value");
        }
        
        // Calculate area with overflow check
        double area = PI * radius * radius;
        
        if (Double.isInfinite(area)) {
            throw new ArithmeticException("Radius too large, calculation overflow");
        }
        
        return area;
    }
    
    public static void main(String[] args) {
        // Test cases
        System.out.println("Test Case 1 - Radius: 5.0");
        System.out.println("Area: " + calculateCircleArea(5.0));
        
        System.out.println("\\nTest Case 2 - Radius: 10.5");
        System.out.println("Area: " + calculateCircleArea(10.5));
        
        System.out.println("\\nTest Case 3 - Radius: 0.0");
        System.out.println("Area: " + calculateCircleArea(0.0));
        
        System.out.println("\\nTest Case 4 - Radius: 1.0");
        System.out.println("Area: " + calculateCircleArea(1.0));
        
        System.out.println("\\nTest Case 5 - Radius: 7.5");
        System.out.println("Area: " + calculateCircleArea(7.5));
        
        // Test error case
        try {
            System.out.println("\\nTest Case 6 - Radius: -5.0 (Error case)");
            System.out.println("Area: " + calculateCircleArea(-5.0));
        } catch (IllegalArgumentException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}
